datafusion/macros/src/lib.rs

```diff
@@ -235,27 +248,23 @@ pub fn user_doc(args: TokenStream, input: TokenStream) -> TokenStream {
         }
     });
 
-    let lock_name: proc_macro2::TokenStream =
-        format!("{name}_DOCUMENTATION").parse().unwrap();
-
     let generated = quote! {
         #input
 
-        static #lock_name: OnceLock<Documentation> = OnceLock::new();
-
         impl #name {
-
-                fn doc(&self) -> Option<&Documentation> {
-                    Some(#lock_name.get_or_init(|| {
-                        Documentation::builder(DocSection { include: #doc_section_include, label: #doc_section_lbl, description: #doc_section_description },
+            fn doc(&self) -> Option<&datafusion_doc::Documentation> {
+                static DOCUMENTATION: std::sync::LazyLock<datafusion_doc::Documentation> =
```

Comment:
thats interesting, I tried doing the same with `OnceLock`, to move it inside the `doc` method but it didn't work. Surprisingly it works with LazyLock